src/ensembl/stories/entity-viewer/transcripts/transcriptData.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  fetchGene,
+  fetchTranscript
+} from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData';
+
+import { Gene } from 'src/content/app/entity-viewer/types/gene';
+import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
+
+enum FeatureType {
+  Gene = 'Gene',
+  Transcript = 'Transcript',
+  Unknown = 'Unknown'
+}
+
+export const getTranscriptData = async (
+  id: string
+): Promise<Gene | Transcript | null> => {
+  const featureType = await getFeatureType(id);
+
+  if (featureType === FeatureType.Gene) {
+    return fetchGene(id);
+  } else if (featureType === FeatureType.Transcript) {
+    return fetchTranscript(id);
+  } else {
+    console.log(`${id} is not a valid id`); // eslint-disable-line no-console
```

Comment:
Shouldn't we use `console.error` here instead? If we do so then we won't need the disable comment.

Reply:
✔️